### ISSUE TITLE
Present etcd to TOC

### ISSUE THREAD

**Issue by erinboyd** [cncf/toc#136]
etcd is a distributed, reliable, consistent, key-value store for the most critical data of a distributed system.  It is used by several CNCF projects, particularly Kubernetes, as well as many other projects in the broader cloud-native ecosystem.  Red Hat supports etcd as a part of OpenShift. We do not have plans to productize etcd independently.

The etcd team believes that this project aligns well with all three goals in section 1 of the CNCF Charter by providing a consistent information store for all other cloud microservices.  Per 1(a) etcd is generally distributed and deployed in containers; per 1(b) it supports dynamic management by providing a place to store and retrieve configuration and location information; per 1(c) it provides only consistent data storage, and allows other microservices to wrap its storage rather than implementing their own.  

Given its already wide industry adoption, the team hopes that moving it to CNCF ownership will encourage a broader community to contribute to, maintain, and build on etcd as a foundation for both new and existing cloud-native projects.  

Website: https://etcd.io 
GitHub: https://github.com/coreos/etcd  (soon to move to a new namespace)
License: Apache License, Version 2.0

**Comment by caniszczyk** [comment]
RFC @cncf/toc 

If I don't hear any objections by next week, we can schedule you on Oct 16th if that works

**Comment by xiang90** [comment]
@erinboyd 

We probably should use `etcd` instead of `Etcd` :)

**Comment by purpleidea** [comment]
I'm against moving things into foundations like this. It's a 501(c)(6) not a 501(c)(3) so we need additional written assurances to prevent complete corporate control and "pay to play" direction so that smaller parties have an influence too.

In particular, if contributing to etcd in the future would require signing a CLA, I would most likely find it difficult to send any future patches.

I'm only contributor #30 ( https://github.com/coreos/etcd/graphs/contributors ) but I'd like to think I'm involved a lot in the project, and have a lot of future work that I'd plan to contribute, as well as do a good deal of testing.

Thanks

**Comment by caniszczyk** [comment]
@purpleidea requiring a CLA is up to the project, CNCF allows projects to choose DCO+Apache2 or CLA+Apache2, it would be up to the etcd community to make that decision.

@purpleidea 501(c) is a tax status and nothing more... there is no pay to play here, it's more PAY TO SUSTAIN the project by having us provide services such as events, technical writing, infrastructure (see https://github.com/cncf/servicedesk)

<img width="1069" alt="screen shot 2018-07-27 at 2 54 38 pm" src="https://user-images.githubusercontent.com/63777/43348078-17be3620-91ad-11e8-88bc-7abce9140ae8.png">

Each project defines its governance and its required to be open, fair and transparent. You can see our technical principles here: https://github.com/cncf/toc/blob/master/PRINCIPLES.md

If you want to discuss more, I'm more than happy to take a call with you to clear up misconceptions: http://calendly.com/caniszczyk

**Comment by purpleidea** [comment]
@caniszczyk 

> DCO+Apache2 or CLA+Apache2

Does CNCF allow projects to choose just Apache2, or LGPLv3+ or AGPLv3+ ?

> 501(c) is a tax status

https://en.wikipedia.org/wiki/501(c)_organization

(c)(3): benefits the public
(c)(6): benefits the paying business membership.

CNCF is the latter. I'm not against business, I'm actually for it, but since there are many heavyweight partners in CNCF it would be great to have some community assurances that those without large memberships in CNCF could have a fair say. Even LF doesn't have a community position any more AIUI.

I believe outside community influences are very valuable, because they help "keep things honest" when paid business coders are often making decisions at the behest of their individual business interests.

So in short, I have a lot of reservations about whether this would be a good idea. If it is, then it's up to you to convince me, otherwise you'll just loose another contributor. (But don't worry, I'm only # 30, so it's not a major loss.)

Thanks!

**Comment by bgrant0607** [comment]
@purpleidea How is owned by one company better with respect to your concerns than owned by a foundation?

**Comment by bgrant0607** [comment]
@caniszczyk @erinboyd I am in favor and am happy to sponsor if there isn't one already

**Comment by erinboyd** [comment]
@bgrant0607 Thank you.

**Comment by caniszczyk** [comment]
@bgrant0607 I'm here to serve so sure, it's a @cncf/toc call.

We can do as soon as Aug 7th or 21st

**Comment by purpleidea** [comment]
@caniszczyk 

You didn't answer my first question. It was:

> Does CNCF allow projects to choose just Apache2, or LGPLv3+ or AGPLv3+ ?

RE:

> @purpleidea How is owned by one company better with respect to your concerns than owned by a foundation?

Because I trust that company to provide good stewardship more than I do the companies controlling the CNCF.

**Comment by caniszczyk** [comment]
@purpleidea 

We default to the Apache v2.0 license but the charter allows other licenses if approved by the board, I linked to our organization charter and IP Policy which covers this information: https://www.cncf.io/about/charter/

I encourage you to take up any issues you have with the etcd official maintainers as they work with the process bring their project under neutral ownership and governance.

re: CNCF, we are a member driven organization that fosters and sustains an ecosystem of open source, vendor-neutral and independently operated projects. You can learn more about our principles here: https://github.com/cncf/toc/blob/master/PRINCIPLES.md

If you want to learn more about the organization and any details about our community as I think you're misunderstanding with how our community and projects work, you can schedule time with me here and I'm happy to work with you and answer any questions: http://calendly.com/caniszczyk - I encourage you to read our annual community report from last year: https://www.cncf.io/cncf-annual-report-2017/

**Comment by warmchang** [comment]
@purpleidea ,I read the blog [(copyleft)](https://purpleidea.com/blog/2017/10/17/copyleft-is-dead-long-live-copyleft/) and learned that you want to adopt the GPL-like license's motivation.

I don't choose which side of the opinion, just hope that etcd can continue to develop better, after all, it is an excellent and widely used software.

**Comment by purpleidea** [comment]
@caniszczyk 

> We default to the Apache v2.0 license but the charter allows other licenses if approved by the board

I believe that a mono-culture of permissively licensed software is bad for the world, and since it's well-known that CNCF and it's membership are strongly against copyleft, then you understand one reason I'm against etcd moving there. Historically, Red Hat has not been against copyleft, which is why I think they're currently a more responsible steward.

If you'd like to dispute the anti-copyleft nature of the CNCF, please respond with some data about which major projects of theirs use copyleft and which don't.

Thanks

**Comment by jdumars** [comment]
FWIW, I am strongly in favor of this proposal.

**Comment by caniszczyk** [comment]
@purpleidea your discussion is out of topic for this particular issue which is around bringing etcd to a neutral home, if you have issues with their decision I encourage you to reach out to the etcd community, for other topics, I already linked my calendly and you can setup a meeting with me any time

I'm going to close this issue and schedule etcd to present at the TOC meeting on Aug 7th or 21st

**Comment by gyuho** [comment]
> I don't want to discuss privately on your "calendly" because open, public discussion is needed, not more private meetings.

I completely agree. And I did not mean to block any discussions (I apologize if you felt that way!).

> non-corporate "community" aspect of open source and etcd if there ever even was one.

I'd really like to know more about your concerns, and let's discuss more in https://github.com/coreos/etcd/issues/9967.
